inference-engine/cmake/plugins/plugins.cmake

```diff
@@ -76,10 +76,11 @@ function(ie_add_plugin)
     add_cpplint_target(${IE_PLUGIN_NAME}_cpplint FOR_TARGETS ${IE_PLUGIN_NAME} CUSTOM_FILTERS ${custom_filter})
 
     # append plugin to the list to register
-
-    list(APPEND PLUGIN_FILES "${IE_PLUGIN_DEVICE_NAME}:${IE_PLUGIN_NAME}")
-    list(REMOVE_DUPLICATES PLUGIN_FILES)
-    set(PLUGIN_FILES "${PLUGIN_FILES}" CACHE INTERNAL "" FORCE)
+    if (BUILD_SHARED_LIBS)
```

Comment:
maybe we call leave this as is, but don't call `ie_register_plugins`?

Reply:
I think we can use plugins passed to `ie_register_plugins` to create a map of function pointers to plugin's `CreatePluginEngine_xxx` functions..

But it can be done in a separate PR, right?

Reply:
I intend to add more change to this PR (see above description)